sql/catalyst/src/main/scala/org/apache/spark/sql/util/SchemaUtils.scala

```diff
@@ -297,4 +296,117 @@ private[spark] object SchemaUtils {
       .replaceAll("\u000B", "\\\\v")
       .replaceAll("\u0007", "\\\\a")
   }
+
+  /**
+   * Check whether the given schema contains a column of the required data type.
```

Comment:
I don't have a strong preference, but it seems a bit weird to define methods in SQL but only call them in MLlib.

Reply:
> I don't have a strong preference, but it seems a bit weird to define methods in SQL but only call them in MLlib.

Yes, but it seems more weird  to have a SchemaUtils in the MLiib too. Do you have any good advice?

Reply:
-1

also feel weird since those utils are only dedicated for ML.

Reply:
Maybe we can rename the `SchemaUtils` in MLlib so that the name is more ML specific?

Reply:
I don't feel strongly about it. WDYT @srowen @WeichenXu123 ?

Reply:
Overall I'm not sure it simplifies things, just moves them around. I'd be neutral on this